source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -285,6 +285,11 @@ void SslSocket::shutdownSsl() {
   if (info_->state() != Ssl::SocketState::ShutdownSent &&
       callbacks_->connection().state() != Network::Connection::State::Closed) {
     int rc = SSL_shutdown(rawSsl());
+    if (rc == 0) {
```

Comment:
Only do this for emulatedEdge Events

Reply:
fixed

Reply:
Please protect with:
if constexpr ( PlatformDefaultTriggerType == FileTriggerType::EmulatedEdge) {

And add a comment explaining why it's needed.